hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsBlobClient.java

```
@@ -1086,6 +1192,75 @@ public String decodeAttribute(byte[] value) throws UnsupportedEncodingException
+    Path pendingJsonPath = new Path(path.getParent(),
+        path.toUri().getPath() + RenameAtomicity.SUFFIX);
+    try {
+      pendingJsonFileStatus = getPathStatus(
+          pendingJsonPath.toUri().getPath(),
+          false, tracingContext, null);
+    } catch (AbfsRestOperationException ex) {
+      if (ex.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+        return;
+      }
+      throw ex;
+    }
+    if (pendingJsonFileStatus != null) {
+      boolean renameSrcHasChanged;
+      try {
+        RenameAtomicity renameAtomicity = getRedoRenameAtomicity(
+            pendingJsonPath, Integer.parseInt(pendingJsonFileStatus.getResult()
+                .getResponseHeader(HttpHeaderConfigurations.CONTENT_LENGTH)),
+            tracingContext);
+        renameAtomicity.redo();
+        renameSrcHasChanged = false;
+      } catch (AbfsRestOperationException ex) {
+        if (ex.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND
+            || ex.getStatusCode() == HttpURLConnection.HTTP_CONFLICT) {
+          renameSrcHasChanged = true;
+        } else {
+          throw ex;
+        }
+      }
+      if (!renameSrcHasChanged) {
+        throw new AbfsRestOperationException(
+            AzureServiceErrorCode.PATH_NOT_FOUND.getStatusCode(),
+            AzureServiceErrorCode.PATH_NOT_FOUND.getErrorCode(),
+            "Path had to be recovered from atomic rename operation.",
+            null);
+      }
+    }
+  }
+
+  @Override
+  public void takeListPathAtomicRenameKeyAction(final Path path,
+      final int renamePendingJsonLen, final TracingContext tracingContext) throws IOException {
+    try {
+      RenameAtomicity renameAtomicity
+          = getRedoRenameAtomicity(path, renamePendingJsonLen, tracingContext);
+      renameAtomicity.redo();
+    } catch (AbfsRestOperationException ex) {
+      if (ex.getStatusCode() != HttpURLConnection.HTTP_NOT_FOUND
+          && ex.getStatusCode() != HttpURLConnection.HTTP_CONFLICT) {
+        throw ex;
```

What about other status code, do we swallow the exception for them ?

---

Have added javadocs for why these two status have to be consumed. 

Reason:
At this point, the path that the json points to can be removed or could have been modified by other thread, in that case, no action gets taken, and no exception has to be thrown.